scipy/stats/_censored_data.py

```diff
@@ -0,0 +1,367 @@
+        raise ValueError('`x` must not contain nan or inf.')
+    if censored.size != x.size:
+        raise ValueError('`x` and `censored` must have the same length.')
+    return x, censored.astype(bool)
+
+
+class CensoredData:
+    """
+    Instances of this class represent censored data.
+
+    Instances may be passed to the ``fit`` method of continuous
+    univariate SciPy distributions for maximum likelihood estimation.
+    The *only* method of the univariate continuous distributions that
+    understands `CensoredData` is the ``fit`` method.  An instance of
+    `CensoredData` can not be passed to methods such as ``pdf`` and
+    ``cdf``.
+
+    An observation is said to be *censored* when the precise value is unknown,
+    but it has a known upper and/or lower bound.  The conventional terminology
+    is:
+
+    * left-censored: an observation is below a certain value but it is
+      unknown by how much.
+    * right-censored: an observation is above a certain value but it is
+      unknown by how much.
+    * interval-censored: an observation lies somewhere on an interval between
+      two values.
+
+    Left-, right-, and interval-censored data can be represented by
+    `CensoredData`.
+
+    For interval-censored data, pass the lower and upper limits of
+    the intervals as one-dimensional arrays to the `CensoredData`
+    constuctor.  By using the values `np.inf` and `-np.inf`, left-
+    and right-censored data may also be created with the constructor.
+
+    For convenience, the class methods ``left_censored`` and
+    ``right_censored`` are provided to create a `CensoredData`
+    instance from a single one-dimensional array of measurements
+    and a corresponding boolean array to indicate which measurements
+    are censored.
+
+    Parameters
+    ----------
+    x : array_like, 1D
+        Uncensored observations.
+    left : array_like, 1D
+        Left-censored observations.
+    right : array_like, 1D
+        Right-censored observations.
+    intervals : array_like, 2D, with shape (m, 2)
+        Interval-censored observations.  Each row ``intervals[k, :]``
+        represents the interval for the kth interval-censored observation.
+
+    Notes
+    -----
+    In the input array `intervals`, the lower bound of the interval may
+    be ``-inf``, and the upper bound may be ``inf``, but at least one must be
+    finite). When the lower bound is ``-inf``, the row represents a left
+    censored observation, and when the upper bound is ``inf``, the row
+    represents a right-censored observation.  If the length of an interval
+    is 0 (i.e. ``intervals[k, 0] == intervals[k, 1]``, the observation is
+    treated as uncensored.  Consequently, one can represent all the types
+    of censored and uncensored data in ``intervals``, but it is generally
+    more convenient to use `x`, `left` and `right` for uncensored,
+    left-censored and right-censored observations, respectively.
+
+    Examples
```

Comment:
I might be missing an example with a distribution.

Reply:
I think this was resolved by f6e7c525ab03794a06ec3cbd0797cccbe745bfdf